core/ledger/blockchain.go

```diff
@@ -37,23 +40,32 @@ func NewBlockchainWithGenesisBlock() (*Blockchain, error) {
 		return nil, NewDetailErr(err, ErrNoCode, "[Blockchain], InitLevelDBStoreWithGenesisBlock failed.")
 	}
 	blockchain := NewBlockchain(height)
+	blockchain.blockSaveCompletedSubscriber = blockchain.BCEvents.Subscribe(events.EventBlockSaveCompleted, blockchain.BlockSaveCompleted)
+
 	return blockchain, nil
 }
 
-func (bc *Blockchain) AddBlock(block *Block) error {
+func (bc *Blockchain) BlockSaveCompleted(v interface{}) {
```

Comment:
Why a new subscriber be added within the module,  looks it do nothing except send a notification,  why not send the EventBlockPersistCompleted notification directly after the  EventBlockSaveCompeleted?